gap/examples.gi

```diff
@@ -370,3 +370,51 @@ GeneralisedPetersenGraphCons);
 
 InstallMethod(GeneralisedPetersenGraph, "for integer, integer", [IsInt, IsInt],
 {n, k} -> GeneralisedPetersenGraphCons(IsImmutableDigraph, n, k));
+
+InstallMethod(KnightsGraphCons,
+"for IsMutableDigraph and two positive integers",
+[IsMutableDigraph, IsPosInt, IsPosInt],
+function(filt, n, k)
+  local D, moveOffSets, coordinates, targetCoordinate, p, i, j;
+  D := EmptyDigraph(IsMutableDigraph, n * k);
+  moveOffSets := [[2, 1], [-2, 1], [2, -1], [-2, -1],
+  [1, 2], [-1, 2], [1, -2], [-1, -2]];
+  coordinates := [];
+  p := 0;
+
+  for i in [1 .. k] do
+    for j in [1 .. n] do
+      p := p + 1;
+      coordinates[p] := [i, j];
+    od;
+  od;
+
+  for i in coordinates do
+    for j in moveOffSets do
+      targetCoordinate := [i[1] + j[1], i[2] + j[2]];
+      if targetCoordinate in coordinates then
+        DigraphAddEdge(D, [Position(coordinates, i),
+        Position(coordinates, targetCoordinate)]);
+      fi;
+    od;
+  od;
```

Comment:
This looks like a fairly inefficient way of doing things, although I suppose we don't need to be too worried about the performance of these named-digraph creation functions 🙂 

In particular, you shouldn't need to search in `coordinates` for each `targetCoordinate`, because you know that `[a,b] in coordinates` if and only if `a in [1 .. k] and b in [1 .. n]`.

Also, you shouldn't need to search for the position of `i` in `coordinates` (because you're selecting `i` from `coordinates` in the first instance, so you could keep track of where it is when you select it), and you shouldn't need to search for `targetCoordinate` in `coordinates` because you could use my remark about `QuotientRemainder` to compute the position directly (by multiplying and adding).

Reply:
Ok, thanks for the tip, that seems like a good idea! The new version doesn't search through coordinates anymore.